### PR TITLE
Update the index page

### PR DESCRIPTION
**Pull request by rkratky**
* Replace the placeholder intro text w/ a real intro.
* Add links to issue tracker and localisation to the Contribution section.
* Reformat the links.

(Fixes [FR-5719](https://warthogs.atlassian.net/browse/FR-5719))

[FR-5719]: https://warthogs.atlassian.net/browse/FR-5719?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -1,28 +1,19 @@
...
-A single sentence that says what the product is, succinctly and memorably
-consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
-dolore magna aliqua.
+Subiquity is an installer framework for Ubuntu. It provides the Ubuntu Server with a text-based installation user interface and Ubuntu Core first-boot configuration.
 
-A paragraph of one to three short sentences, that describe what the product
-does. Urna cursus eget nunc scelerisque viverra mauris in. Nibh mauris cursus
-mattis molestie a iaculis at vestibulum rhoncus est pellentesque elit. Diam
-phasellus vestibulum lorem sed.
+This documentation set offers guidance for installing Ubuntu Server using Subiquity.
```
Comment (dbungert) [inline doc/index.rst:8]: This is also the home of documentation covering autoinstall for any case, and it is not implausible that Desktop facing documentation may end up here also.

File: doc/index.rst [hunk 2]
```diff
@@ -1,28 +1,19 @@
...
 
-Ubuntu Installation documentation
-#################################
+Ubuntu installation guide
+=========================
 
-A single sentence that says what the product is, succinctly and memorably
-consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
-dolore magna aliqua.
+Subiquity is an installer framework for Ubuntu. It provides the Ubuntu Server with a text-based installation user interface and Ubuntu Core first-boot configuration.
```
Comment (dbungert) [inline doc/index.rst:6]: Subiquity is the backend for ubuntu-desktop-installer, a short mention here sounds wise.

File: doc/index.rst (hunk 1 above)
Comment (rkratky) [reply doc/index.rst:8]: I tried to cover that in the note below (seeing that the desktop stuff isn't here yet). 

I'll add info about autoinstall.

File: doc/index.rst (hunk 2 above)
Comment (rkratky) [reply doc/index.rst:6]: Ack

**Review by dbungert — APPROVED** [review]

**Pull request merged**